rocAL/include/loaders/audio/audio_loader.h

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "audio_read_and_decode.h"
+#include "circular_buffer.h"
+#include "commons.h"
+#include "meta_data_reader.h"
+
+#ifdef ROCAL_AUDIO
+
+// AudioLoader runs an internal thread for loading and decoding of audios asynchronously
+// It uses a circular buffer to store decoded audios for the user
+class AudioLoader : public LoaderModule {
+   public:
+    explicit AudioLoader(void* dev_resources);
+    ~AudioLoader() override;
+    LoaderModuleStatus load_next() override;
+    void initialize(ReaderConfig reader_cfg, DecoderConfig decoder_cfg, RocalMemType mem_type, unsigned batch_size, bool keep_orig_size = false) override;
+    void set_output(Tensor* output_audio) override;
+    size_t remaining_count() override;  // returns number of remaining items to be loaded
+    void reset() override;              // Resets the loader to load from the beginning of the media
+    Timing timing() override;
+    void start_loading() override;
+    LoaderModuleStatus set_cpu_affinity(cpu_set_t cpu_mask);
+    LoaderModuleStatus set_cpu_sched_policy(struct sched_param sched_policy);
```

Comment:
Are we using these functions in audio?

Reply:
Yes, They are being used here [link](https://github.com/swetha097/rocAL/blob/3a865078362d764f3ce00f738e531c071cfccf54/rocAL/source/loaders/audio/audio_loader_sharded.cpp#L91)